query.js

```diff
@@ -293,12 +293,35 @@ var searchMap = {
                     { travelEndMonth: { $exists: false } },
                     { travelStartMonth: { $eq: 0 } },
                     { travelEndMonth: { $eq: 0 } },
-                    {
+                    { 
+                        $and: [
+                            { travelEndYear: { $gt: startYear } },
+                            { travelStartYear: { $lt: startYear } }
+                        ],
+                    },
+                    { 
+                        $and: [
+                            { travelEndYear: { $eq: startYear } },
+                            { travelStartYear: { $eq: startYear } },
```

Comment:
travelStartYear = startYear = travelEndYear

then compare if startMonth is between travelStartMonth and travelEndMonth